src/pages/index.tsx

```diff
@@ -12,14 +29,134 @@ import EditorErrorBoundary from '~components/errorBoundaries/EditorErrorBoundary
 import Editor from '~components/editor/Editor'
 import { InspectorProvider } from '~contexts/inspector-context'
 import Inspector from '~components/inspector/Inspector'
+import { checkUser, createProject } from '~utils/checkProject'
+import { useSelector } from 'react-redux'
+import { signIn, useSession } from 'next-auth/client'
+import { getComponents } from '~core/selectors/components'
+import { useRouter } from 'next/router'
 
 interface Props {
   id: number
   loading: boolean
 }
 
+interface Project {
+  createdAt: string
+  updatedAt: string
+  userId: number
+  id: number
+  markup: string
+  projectName: string
+}
+
 const App = (props: Props) => {
   const { handlers } = useShortcuts()
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [projectName, setProjectName] = useState('')
+  const components = useSelector(getComponents)
+  const [session] = useSession()
+  const router = useRouter()
+  const toast = useToast()
+
+  const updateProject = async () => {
+    const markup = JSON.stringify(components)
+    let bodyData = {
+      project: {
+        markup: markup,
+        id: props.id,
+      },
+    }
+    const response = await fetch('/api/project/update', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(bodyData),
+    })
+    const data = await response.json()
+    return data
+  }
+
+  const initProject = async () => {
+    if (projectName.length > 0) {
+      const markup = JSON.stringify(components)
+      let newProject = await createProject(markup, projectName)
+      if (newProject) {
+        toast({
+          title: 'Created project',
+          description: 'The project has been created successfully',
+          status: 'success',
+          duration: 9000,
+          isClosable: true,
+        })
+        const href = `/project/${newProject.id}-${newProject.projectName}`
+        router.push(href, href, { shallow: true })
+      } else {
+        toast({
+          title: 'Error when created project',
+          description: 'An error occured, try again later',
+          status: 'error',
+          duration: 9000,
+          isClosable: true,
+        })
+      }
+    } else {
+      toast({
+        title: 'Please enter a valid project name',
+        description: 'Project name cannot be empty',
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      })
+    }
+  }
+
+  const saveProject = async () => {
+    if (session) {
+      if (props.id) {
+        let userCanEdit = false
+        const userProject = await checkUser(session.user.name)
+        userProject.project.map((e: Project) => {
```

Comment:
You can use the some function here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

const userCanEdit = userProject.project.some((e: Project) => (e.id === props.id))